src/App.tsx

```diff
@@ -1,24 +1,246 @@
+    setIsDeletingCompleted(true);
+    try {
+      const completedTodos = todos.filter(todo => todo.completed);
+
+      await Promise.all(completedTodos.map(todo => deleteTodo(todo.id)));
+
+      setTodos(prevTodos => prevTodos.filter(todo => !todo.completed));
+    } catch (error) {
+      setErrorMessage(t('Error.delete'));
+    } finally {
+      setIsDeletingCompleted(false);
+    }
+  }, [todos]);
+  // #endregion
+
+  // #region Update functions
+  const updateTodo = useCallback(async (
+    id: number,
+    data: Partial<Todo>,
+  ): Promise<void> => {
+    try {
+      const updetedTodo = await patchTodo(id, data);
+
+      setTodos(prevTodos => prevTodos.map(todo => (
+        todo.id !== updetedTodo.id
+          ? todo
+          : updetedTodo
+      )));
+    } catch (error) {
+      setErrorMessage(t('Error.update'));
+    }
+  }, []);
+
+  const updateAllTodosComplete = useCallback(async (): Promise<void> => {
+    setIsUpdatingAllTodo(true);
+    try {
+      await Promise.all(todos.map(todo => {
+        const updatedTodo = {
+          ...todo,
+          completed: !toggleStatus,
+        };
+
+        return updateTodo(todo.id, updatedTodo);
+      }));
+
+      setTodos(prevTodos => prevTodos.map(todo => ({
+        ...todo,
+        completed: !toggleStatus,
+      })));
+
+      setToggleStatus(!toggleStatus);
+    } catch (error) {
+      setErrorMessage(t('Error.update'));
+    } finally {
+      setIsUpdatingAllTodo(false);
+    }
+  }, [todos, toggleStatus]);
+  // #endregion
+
+  const headerContextValue = useMemo(() => ({
+    setErrorMessage,
+    uploadTodo,
+  }), [setErrorMessage, uploadTodo]);
+
+  const fetchContextValue = useMemo(() => ({
+    deleteTodos,
+    updateTodo,
+  }), [deleteTodos, updateTodo]);
+
+  const footerContextValue = useMemo(() => ({
+    notCompletedTodoCount,
+    setActiveFilter,
+    activeFilter,
+    isCompletedExist,
+    deleteCompletedTodos,
+  }), [
+    notCompletedTodoCount,
+    activeFilter,
+    isCompletedExist,
+  ]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <LanguageSelection />
+
+      <div className="todoapp__content">
+        <HeaderContext.Provider value={headerContextValue}>
+          <Header
+            todos={todos}
+            toggleStatus={toggleStatus}
+            onUpdateAllTodosComplete={updateAllTodosComplete}
+          />
+        </HeaderContext.Provider>
+
+        {visibleTodos && (
+          <section className="todoapp__main">
+            <FetchContext.Provider value={fetchContextValue}>
```

Comment:
Conditionally rendering the TodoList component with 'visibleTodos &&' is not necessary. You can remove this check, as 'visibleTodos' will always be an array.